packages/block-library/src/navigation/edit/unsaved-inner-blocks.js

```diff
@@ -51,13 +91,26 @@ export default function UnsavedInnerBlocks( {
 		}
 	}, [ blocks ] );
 
-	// If the current inner blocks object is different in any way
-	// from the original inner blocks from the post content then the
-	// user has made changes to the inner blocks. At this point the inner
-	// blocks can be considered "dirty".
-	// We also make sure the current innerBlocks had a chance to be set.
-	const innerBlocksAreDirty =
-		!! originalBlocks.current && blocks !== originalBlocks.current;
+	// If the current inner blocks is different  from the original inner blocks
+	// from the post content then then the user has made changes to the inner blocks.
+	// At this point the inner blocks can be considered "dirty".
+	// Note: referential equality is not sufficient for comparison as the inner blocks
+	// of the page list are controlled and may be updated async due to syncing with
+	// entity records. As a result we need to perform a deep equality check skipping
+	// the page list's inner blocks.
+	const innerBlocksAreDirty = ! isDeepEqual(
+		originalBlocks.current,
+		blocks,
+		( prop, x ) => {
+			// Skip inner blocks of page list during comparison as they
```

Comment:
Does this actually skip innerBlocks for all blocks, not just pageList?

Reply:
Yes and it should be great right? :)

Reply:
Except for the comment :)

Reply:
Hmmm no I think @scruffian it right. Most other types of blocks that use inner blocks are writeable where as Page List is specifically read only.

Not sure how we can foresee all cases so I suggest we optimise for `core/page-list` only at this point.

Reply:
See https://github.com/WordPress/gutenberg/pull/46223/files#r1040889011